ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -898,14 +962,28 @@ int mca_pml_ucx_isend(const void *buf, size_t count, ompi_datatype_t *datatype,
         return OMPI_ERROR;
     }
 
+    if (OPAL_LIKELY(OMPI_COMM_IS_GLOBAL_INDEX(comm))) {
```

Comment:
need to move that code to separate function and call it from mca_pml_ucx_isend and mca_pml_ucx_isend_init.